app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java

```diff
@@ -59,13 +67,21 @@ public Result getSubpopulation(String guid) {
         Subpopulation subpop = subpopService.getSubpopulation(session.getStudyIdentifier(), subpopGuid);
         return okResult(subpop);
     }
-    public Result deleteSubpopulation(String guid) {
-        UserSession session = getAuthenticatedSession(DEVELOPER);
+    public Result deleteSubpopulation(String guid, String physicalDeleteString) {
+        UserSession session = getAuthenticatedSession();
+        if (!session.getUser().isInRole(DELETE_ROLES)) {
+            throw new UnauthorizedException();
+        }
+        
         SubpopulationGuid subpopGuid = SubpopulationGuid.create(guid);
         
-        subpopService.deleteSubpopulation(session.getStudyIdentifier(), subpopGuid);
+        // Only admins can request a physical delete.
+        boolean physicalDelete = (session.getUser().isInRole(ADMIN) && "true".equals(physicalDeleteString));
```

Comment:
changed. I know a separate URL would be another option but this leads to encoding roles in URLs, and we've already shed some blood to eliminate that. The authorization should be encapsulated behind the endpoint.